File: src/frontend/src/app/components/base/model/page-request.ts
import { FilterDTO } from './filter-dto';

export class PageRequest {
  filter: FilterDTO;
  page: number;
  size: number;
  order: Order[];

  constructor(filter: FilterDTO, size: number, page: number, sort: Order[]) {
    this.filter = filter;
    this.page = page;
    this.size = size;
    this.order = sort;
  }
}

export interface Pageable {
  page: number;
  size: number;
  sort: string | null;
}

export interface Order {
  direction: Direction;
  property: string;
}

export enum Direction {
  ASC,
  DESC,
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/core/dto/PageRequest.java
package br.com.grupopipa.gestaointegrada.core.dto;

import java.util.List;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class PageRequest {
    private FilterDTO filter;
    private Integer page;
    private Integer size;
    private List<OrderDTO> order;
}
